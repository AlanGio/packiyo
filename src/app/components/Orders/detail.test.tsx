import React from "react";
import { render } from "@testing-library/react";
import OrderDetail from "./detail";

describe("OrderDetail", () => {
  const orderData = {
    attributes: {
      number: "12345",
      status_text: "Pending",
      shipping: 10,
      tax: 2,
      discount: 5,
      total: 100,
      ordered_at: "2022-01-01T00:00:00Z",
      updated_at: "2022-01-02T00:00:00Z",
      packing_note: "Handle with care",
      shipping_method_name: "Standard",
      tags: ["tag1", "tag2"],
    },
    relationships: {
      shipping_contact_information: {
        data: { id: "John Doe" },
      },
      billing_contact_information: {
        data: { id: "Jane Smith" },
      },
      order_items: {
        data: [{ id: "item1" }, { id: "item2" }],
      },
    },
  };

  test("renders order details correctly", () => {
    const { getByText } = render(<OrderDetail orderData={orderData} />);

    expect(getByText("Order Number: 12345")).toMatchSnapshot();
    expect(getByText("Status: Pending")).toMatchSnapshot();
    expect(getByText("Shipping: $10")).toMatchSnapshot();
    expect(getByText("Tax: $2")).toMatchSnapshot();
    expect(getByText("Discount: $5")).toMatchSnapshot();
    expect(getByText("Total: $100")).toMatchSnapshot();
    expect(getByText("Packing Note: Handle with care")).toMatchSnapshot();
    expect(getByText("Shipping Method: Standard")).toMatchSnapshot();
    expect(getByText("Name: John Doe")).toMatchSnapshot();
    expect(getByText("Name: Jane Smith")).toMatchSnapshot();
    expect(getByText("Item ID: item1")).toMatchSnapshot();
    expect(getByText("Item ID: item2")).toMatchSnapshot();
  });

  test("renders 'Order not found' when orderData is null", () => {
    const { getByText } = render(<OrderDetail orderData={null} />);

    expect(getByText("Order not found")).toMatchSnapshot();
  });
});
