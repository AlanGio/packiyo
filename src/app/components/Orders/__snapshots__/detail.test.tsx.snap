// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetail renders 'Order not found' when orderData is null 1`] = `
<h6
  class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
>
  Order not found
</h6>
`;

exports[`OrderDetail renders order details correctly 1`] = `
<h6
  class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
>
  Order Number: 
  12345
</h6>
`;

exports[`OrderDetail renders order details correctly 2`] = `
<h6
  class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
>
  Status: 
  Pending
</h6>
`;

exports[`OrderDetail renders order details correctly 3`] = `
<h6
  class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
>
  Shipping: $
  10
</h6>
`;

exports[`OrderDetail renders order details correctly 4`] = `
<h6
  class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
>
  Tax: $
  2
</h6>
`;

exports[`OrderDetail renders order details correctly 5`] = `
<h6
  class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
>
  Discount: $
  5
</h6>
`;

exports[`OrderDetail renders order details correctly 6`] = `
<h6
  class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
>
  Total: $
  100
</h6>
`;

exports[`OrderDetail renders order details correctly 7`] = `
<h6
  class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
>
  Packing Note: 
  Handle with care
</h6>
`;

exports[`OrderDetail renders order details correctly 8`] = `
<h6
  class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
>
  Shipping Method: 
  Standard
</h6>
`;

exports[`OrderDetail renders order details correctly 9`] = `
<p
  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
>
  Name: 
  John Doe
</p>
`;

exports[`OrderDetail renders order details correctly 10`] = `
<p
  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
>
  Name: 
  Jane Smith
</p>
`;

exports[`OrderDetail renders order details correctly 11`] = `
<p
  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
>
  Item ID: 
  item1
</p>
`;

exports[`OrderDetail renders order details correctly 12`] = `
<p
  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
>
  Item ID: 
  item2
</p>
`;
